.github/workflows/publish-to-npm.yml

```diff
@@ -103,3 +103,41 @@ jobs:
       - run: |
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc
           npm dist-tag add @salesforce/sfdx-scanner@${{ github.event.release.tag_name || inputs.tag }} latest-beta
+  # Step 5: Create a Pull Request for merging `main` into `dev`
+  create-main2dev-pull-request:
+    needs: promote-to-latest
+    runs-on: macos-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      # Check out `main`
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main'
+      # Create a new branch based on `main`, so that merge conflicts can be manually resolved if need be.
+      - run: |
+          NEW_VERSION=$(jq -r ".version" package.json)
+          git checkout -b m2d/v$NEW_VERSION
+          git push --set-upstream origin m2d/v$NEW_VERSION
+      # Create a Pull Request from the new branch into `dev`.
+      - run: |
+          NEW_VERSION=$(jq -r ".version" package.json)
+          # For whatever reason, the version of 'echo' on GHAs doesn't process backspace by default.
+          # The non-POSIX-standard -e flag causes it to do that.
+          echo -e "This branch and PR were automatically created following the successful release of v$NEW_VERSION.\n\
```

Comment:
Request for feedback: Is this explanation clear? Is there anything confusing?

Reply:
This explanation is clear. I am just wondering if instead of manual step being involved here, can we merge to dev automatically if there are no merge conflicts?

Reply:
Our branch protection rules prevent pushing directly to `dev`. Maybe there's a way to merge the pull request automatically as well? I can look into it.

Reply:
@jag-j , there's a `gh pr merge` command. I can run some additional experiments to see if we can use that to meet our needs here.